bin/cheribsdtest/cheribsdtest_malloc.c

```diff
@@ -310,6 +310,60 @@ CHERIBSDTEST(malloc_revocation_ctl_suid_elfnote_enable_protctl_disable,
 	    "malloc_is_revoking_suid_elfnote_enable", true, &arg);
 }
 
+static void
+malloc_early_constructor_common(const char *progname)
+{
+	pid_t pid;
+	int res;
+
+	pid = fork();
+	CHERIBSDTEST_VERIFY(pid >= 0);
+	if (pid == 0) {
+		char *progpath;
+		char *argv[2];
+
+		asprintf(&progpath, "/usr/libexec/%s", progname);
+		argv[0] = progpath;
+		argv[1] = NULL;
+		execve(argv[0], argv, NULL);
+		abort();
+	} else {
+		waitpid(pid, &res, 0);
+		if (WIFEXITED(res) && WEXITSTATUS(res) == 0)
+			cheribsdtest_success();
+		else
+			cheribsdtest_failure_errx("child exited improperly");
+	}
+}
+
+CHERIBSDTEST(malloc_early_constructor_dynamic,
+    "invoke malloc in an early constructor (dynamic binary)")
```

Comment:
Hm, I meant having cheribsdtest-purecap run the purecap static binary, cheribsdtest-purecap-dynamic run the purecap dynamic binary, etc

Reply:
I've now done this. I did notice that the -mt variants are mostly no-ops as a result of @markjdb's async revocation tests in abb906e738781c0391f924e721afcac83c5bed45. I'm not quite sure what we want to do there.

Reply:
Oh. Those should be limited to the -mt builds. The point of having cheribsdtest variants not using threads is to test things like signal delivery that get hooked by libthr. I believe that's done eagerly in a constructor, so even single-threaded binaries get that, and the only way to avoid it is to not link against libthr.